os/mm/umm_heap/umm_sem.c

```diff
@@ -101,8 +102,8 @@ int umm_trysemaphore(void *address)
 	if (heap) {
 		return mm_trysemaphore(heap);
```

Comment:
The mm_trysemaphore returns OK or ERROR, not negated errno. Let's check the result of it and if it fails, let's get errno and return negated errno as description says.